packages/core/src/payment/payment-strategy-action-creator.ts

```diff
@@ -46,14 +49,18 @@ export default class PaymentStrategyActionCreator {
                             throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
                         }
 
-                        strategy = this._strategyRegistry.getByMethod(method);
+                        try {
+                            strategy = this._strategyRegistryV2.get({ id: method.id });
+                        } catch {
+                            strategy = this._strategyRegistry.getByMethod(method)
+                        }
                     } else {
                         strategy = this._strategyRegistry.get(PaymentStrategyType.NO_PAYMENT_DATA_REQUIRED);
                     }
 
-                    return strategy
-                        .execute(payload, { ...options, methodId: payment.methodId, gatewayId: payment.gatewayId })
-                        .then(() => createAction(PaymentStrategyActionType.ExecuteSucceeded, undefined, meta));
+                    const promise: Promise<InternalCheckoutSelectors | void> = strategy.execute(payload, { ...options, methodId: payment.methodId, gatewayId: payment.gatewayId });
```

Comment:
You probably can keep what's there before instead of assigning to a temp variable.

Reply:
I had to do this since typescript was confused about mixed return types since either of the strategy methods had their return types not compatible, so it's essentially setting a type.